click/simplelog.py

```diff
@@ -0,0 +1,199 @@
+                 debug_level=None, logger_name=None, flavor=None):
+
+        """ Handler class for SimpleLog
+
+        :param verbose_help: Override the text displayed for the verbose
+                     parameter [Enable verbose logging]
+        :param debug_help: Override the text displayed for the debug parameter
+                           [Enable debugging]
+        :param quiet_help: Override the text displayed for the quiet parameter
+                           [Enable quiet mode]
+        :param remove_args: Remove the logging arguments, so they are not
+                            submitted to the main command [Disabled]
+        :param verbose_args: List of arguments for the "verbose"-setting
+                             [--verbose and -v]
+        :param debug_args: List of arguments for the "debug"-setting
+                           [--debug and -d]
+        :param quiet_args: List of arguments for the "quiet"-setting
+                           [--quiet and -q]
+        :param verbose_arg_name: Name of the "verbose" argument [verbose]
+        :param debug_arg_name: Name of the "debug" argument [debug]
+        :param quiet_arg_name: Name of the "quiet" argument [quiet]
+        :param default_level: Default level [logging.ERROR]
+        :param verbose_level: Level for "verbose"-setting [logging.INFO]
+        :param debug_level: Level for "debug"-setting [logging.DEBUG]
+        :param quiet_level: Level for "quiet"-setting [logging.CRITICAL]
+        :param logger_name: Logger to configure [root-logger]
+        :param flavor: Parameter-flavor to use:
+                       * qvd: Usage of --quiet, --verbose and --debug parameters
+                       * vvv: Usage mulitple -v parameters (Uses
+                         "Increase verbosity" for verbose_help-parameter and the
+                         verbose_args and verbose_arg_name)
+        """
+
+        self.callback = callback
+
+        self.verbose_help = verbose_help or None
+        self.debug_help = debug_help or "Enable debugging"
+        self.quiet_help = quiet_help or "Enable quiet mode"
+        self.remove_args = remove_args or False
+        self.verbose_args = verbose_args or ["--verbose", "-v"]
+        self.debug_args = debug_args or ["--debug", "-d"]
+        self.quiet_args = quiet_args or ["--quiet", "-q"]
+        self.verbose_arg_name = verbose_arg_name or "verbose"
+        self.debug_arg_name = debug_arg_name or "debug"
+        self.quiet_arg_name = quiet_arg_name or "quiet"
+        self.default_level = default_level or logging.ERROR
+        self.verbose_level = verbose_level or logging.INFO
+        self.debug_level = debug_level or logging.DEBUG
+        self.quiet_level = quiet_level or logging.CRITICAL
+        self.logger_name = logger_name or None
+        self.flavor = flavor or "qvd"
+
+        if self.flavor == "qvd":
+
+            self.verbose_help = "Enable verbose logging"
+
+            self.__click_params__ = [
```

Comment:
That seems like poking at click's internal state... maybe you could simply wrap the callback normally with ``click.option`` etc?